src/main/java/org/opensearch/knn/index/query/KNNQueryBuilder.java

```diff
@@ -94,6 +109,29 @@ public KNNQueryBuilder(String fieldName, float[] vector, int k, QueryBuilder fil
         this.k = k;
         this.filter = filter;
         this.ignoreUnmapped = false;
+        this.radius = 0.0f;
```

Comment:
nit: What do you think about using object instead of primitives? IMO, saying radius is null seems technically correct than radius is 0 for top k

Reply:
It will need both k and radius update to object, which will involve other existing code change for k necessitates additional null checks. What do you think about? @jmazanec15 @navneet1v

Reply:
Makes sense, I'll change radius use object.

Reply:
Updated to use `distance` as object